### DISCUSSION TITLE
Using the terms "BlueALSA" or "bluealsa" in the names of other projects

### DISCUSSION THREAD

**Discussion by borine** [Arkq/bluez-alsa#620] [General]
I'm thinking about sharing some of my BlueALSA related scripts and tools as separate projects in my own repository. The first thing I need to do is choose a name for each project. Given that they all relate to, and depend on, BlueALSA, my first thought is to include "BlueALSA" or "bluealsa" in the name.

However, for both terms the copyright and any other intellectual property rights clearly belong to @arkq. So, @arkq, would you prefer me to use some other terms, or are you OK with me to effectively "steal" some of the excellent reputation built up by the bluez-alsa project?

I already have a project called "bluealsa-monitor", but that is no longer maintained and I will be happy to delete it if required (or re-name it in the unlikely event anyone is using it).

The specific projects I have in mind are:
- a service for creating and removing ALSA configuration entries as bluealsa PCMs connect and disconnect. This one is currently in a branch in my fork of bluez-alsa, but I now think it would be easier to maintain as a stand-alone project. It does require copies of some of the files in the bluez-alsa "src/shared" directory, but these would be clearly marked as having been taken from bluez-alsa. I would share this project under the MIT license to keep things simple.
- The ALSA hook plugin for HFP-AG. Currently this is PR #525, but again I now think it would be more easily maintained as a stand-alone project. Again it requires copies of some of the "src/shared" files, so I would make this MIT licensed too.
- A set of bash scripts that create a simple hands-free device using existing utilities and sample programs from bluez-alsa, ALSA and bluez. This was developed while experimenting with the latest features of BlueALSA.

I would like to make sure that I "do the right thing" to avoid any complications in the future. I am happy to choose different names if that avoids any potential issues.

**Comment by arkq** [comment]
> However, for both terms the copyright and any other intellectual property rights...

I think that you can use both these names if your project is related to bluez-alsa project. It will be easier for anyone to find stuff related to bluez-alsa if names are "interlinked". I do not claim any trade mark for BlueALSA name, though :D

> The specific projects I have in mind are ....

I can see that there will be more than one project, so maaaaaybeeee it would be better to "bundle" all these projects under github organization account? Similar to https://github.com/bluez and others. What do you think? However, there might be one problem with that... we will have to choose a name for such organization, and I suck at naming things :D The only think that comes to my mind is simply bluez-alsa. And then to align namings better the bluez-alsa repo could be renames as bluealsa, so the full link would be https://github.com/bluez-alsa/bluealsa Anyway, all that stuff can be done later. GitHub has got a "transfer repo" feature, so any repo can be transferred to another account.

**Reply by FoulOreo** [reply]
@arkq I was wondering if you could provide a zip file for your latex master plugin so that it can be installed via the load plugin from file option on calibre (I tried using the zip file from github itself but that didn't work - probably cos it wasn't the installation methods you specified but I wasn't sure about them). I wasn't sure how else to contact you so I apologise for doing so on an unrelated discussion page. Im rather new to github and not exactly a pro at using calibre or at much else tbh. I would greatly appreciate your help.

**Reply by seamusdemora** [reply]
Apologies for butting in... I saw this thread under mine & couldn't resist :)   I'm not very good at naming things either, but it seems to me that it **is** important to retain the "brand recognition" that you've built with 'bluez-alsa'.  And so my suggestion for an organizational name to put your repos under is: 'bluez-alsa-org'.  Okay - I'll butt out now, but best wishes to you both.

**Comment by borine** [comment]
I've created three new repos with the projects mentioned above. They are:

* ALSA automated configuration
https://github.com/borine/bluealsa-autoconfig

* HFP-AG helper plugin
https://github.com/borine/bluealsa-hfp-ag-plugin

* HFP-HF tools
https://github.com/borine/bluealsa-simple-handsfree

@arkq I'm not convinced that the engineering quality of these is good enough to be bundled so closely to the bluez-alsa project, but if you feel that any of them would be beneficial to the bluez-alsa project then I would not object to some or all of them being transferred into a organization account.
